src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,37 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle)obj;
+            return ((color == null && rectangle.color == null)
+                        || (color != null && rectangle.color != null)
+                        && rectangle.getColor().equals(getColor()))
+                        && ((width == null && rectangle.width == null)
+                        || (width != null && rectangle.width != null)
+                        && rectangle.getWidth().equals(getWidth()))
+                        && ((length == null && rectangle.length == null)
+                        || (length != null && rectangle.length != null)
+                        && rectangle.getLength().equals(getLength()));
```

Comment:
good job

here is how we can simplify this a bit
note that we don't have to use getters because we are located inside the class

```suggestion
            return (color == rectangle.color              // both null check
                        || (color != null && color.equals(rectangle.color)))
                        && (length == rectangle.length
                        || (length != null && length.equals(rectangle.length)))
                        && (width == rectangle.width 
                        || (width != null && width.equals(rectangle.width)));
```

Reply:
done